Add example for ^mimeType and ^mediaType metadata selectors (#144)
mulesoft/docs-dataweave: modules/ROOT/pages/dataweave-formats-multipart.adoc

}
----

[[example5]]
=== Example: Using ^mediaType and ^mimeType Metadata Selectors

> The `^mimeType` selector returns the MIME type of a value without parameters, for example, `application/json`. +

Reviewer: ```suggestion
The `^mimeType` selector returns the MIME type of a value without parameters, for example, `application/json`. 

```

Reviewer: I don't think we use the `+` to start a new line like that in our docs.

Another option might be to provide some sort of intro sentence and describe `^mimeType` and `^mediaType` below that sentence in separate bullets.

> The `^mediaType` selector by contrast includes any other related parameters related to the MIME type, for example, `application/json;charset=UTF-16`.

Reviewer: ```suggestion
The `^mediaType` selector, by contrast, provides the MIME type and any other parameters related to it, for example, `application/json;charset=UTF-16`.
```

The following example shows how to transform multipart inputs into multipart outputs with a different format. The DataWeave script uses both the `^mimeType` and `^mediaType` selectors to generate multipart content inside other multipart content.

=== Input
The multipart input has Content-Type `application/json`:

[source,text,linenums]
----
----------------------------049709565842371278701691
Content-Type: application/json
Content-Disposition: form-data; name="order"

{
	"name": "order1",
	"id": 123,
	"content": "orderContent"
}
----------------------------049709565842371278701691
Content-Type: application/json
Content-Disposition: form-data; name="partner"

{
 "name": "mulesoft",
 "id": 1
}
----------------------------049709565842371278701691--
----

=== Source

The DataWeave script transforms the multipart input into multipart output by defining `output multipart/form-data` MIME type. The multipart output data contains three parts:

> * `orderAck` +

Reviewer: ```suggestion

* `orderAck` +
```

Reviewer: need blank line above the bullets

> Contains a text with an acknowledge of the receipt of an order. The Content-Type is `text/plain`.

Reviewer: ```suggestion
Contains a text with an acknowledgement of the receipt of an order. The Content-Type is `text/plain`.
```

* `order` +
Contains the order itself. The Content-Type is `payload.parts.order.content.^mimeType`. The ^mimeType` selector becomes useful to generate the Content-Type field.
* `originalContent`

> Contains the original content which is multipart. The Content-Type is `payload.^mediaType`. The ^mediaType selector includes useful metadata such as the original boundary.

Reviewer: ```suggestion
Contains the original content, which is multipart. The Content-Type is `payload.^mediaType`. The ^mediaType selector includes useful metadata such as the original boundary.
```
